addons/binding/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/handler/FoobotHandler.java

```diff
@@ -0,0 +1,340 @@
+                }
+
+            } catch (MalformedURLException e) {
+                errorMsg = e.getMessage();
+                logger.warn("Constructed url is not valid: {}", errorMsg);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+            } catch (JsonSyntaxException e) {
+                errorMsg = "Configuration is incorrect";
+                logger.warn("Error running foobot.io request: {}", errorMsg);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+            } catch (IOException | IllegalStateException e) {
+                errorMsg = e.getMessage();
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+            startAutomaticRefresh();
+        } else
+
+        {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+        }
+    }
+
+    /**
+     * Start the job refreshing the Sensor data from Foobot device
+     */
+    private void startAutomaticRefresh() {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        // Request new sensor data from foobot.io service
+                        foobotData = updateFoobotData();
+
+                        // Update all channels from the updated Sensor data
+                        for (Channel channel : getThing().getChannels()) {
+                            updateChannel(channel.getUID().getId(), foobotData);
+                        }
+                    } catch (Exception e) {
+                        logger.error("Exception occurred during execution: {}", e.getMessage(), e);
+                    }
+                }
+            };
+
+            config = getConfigAs(FoobotConfiguration.class);
+            int delay = (config.refresh != null) ? config.refresh.intValue() : DEFAULT_REFRESH_PERIOD;
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, delay, TimeUnit.MINUTES);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the Foobot handler.");
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(@NonNull ChannelUID channelUID, @NonNull Command command) {
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId(), foobotData);
+        } else {
+            logger.debug("The Foobot binding is read-only and can not handle command {}", command);
+        }
+
+    }
+
+    private void updateChannel(String channelId, FoobotJsonData foobotResponse) {
+        if (isLinked(channelId)) {
```

Comment:
As you already have fetched the data at this point, you can ommit the `isLinked(...)` check here. The framework will know what to do if there is no item linked to any channel.